{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"KG-OntoML.ipynb","provenance":[],"collapsed_sections":[],"authorship_tag":"ABX9TyNvNUCmA9EQX/fd911E0odz"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"},"accelerator":"GPU","widgets":{"application/vnd.jupyter.widget-state+json":{"fc7b9510580d440e8e226aee783a9ee3":{"model_module":"@jupyter-widgets/controls","model_name":"HBoxModel","model_module_version":"1.5.0","state":{"_view_name":"HBoxView","_dom_classes":[],"_model_name":"HBoxModel","_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.5.0","box_style":"","layout":"IPY_MODEL_6def71a8cb2d4717a6abf73d4f8e5265","_model_module":"@jupyter-widgets/controls","children":["IPY_MODEL_ca9a3f2486ae423b900c6f55355f654a","IPY_MODEL_30b7b7b3d0aa4fe49ba756cfce0254e2","IPY_MODEL_18098799052e443baaa6c9e27eaac950"]}},"6def71a8cb2d4717a6abf73d4f8e5265":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","model_module_version":"1.2.0","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}},"ca9a3f2486ae423b900c6f55355f654a":{"model_module":"@jupyter-widgets/controls","model_name":"HTMLModel","model_module_version":"1.5.0","state":{"_view_name":"HTMLView","style":"IPY_MODEL_48ac18cc60b7457db81eee6a2e4984a5","_dom_classes":[],"description":"","_model_name":"HTMLModel","placeholder":"​","_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","value":"  0%","_view_count":null,"_view_module_version":"1.5.0","description_tooltip":null,"_model_module":"@jupyter-widgets/controls","layout":"IPY_MODEL_46e55f0c552547d4b543b1ccfb0ac2b7"}},"30b7b7b3d0aa4fe49ba756cfce0254e2":{"model_module":"@jupyter-widgets/controls","model_name":"FloatProgressModel","model_module_version":"1.5.0","state":{"_view_name":"ProgressView","style":"IPY_MODEL_6f25657095c4435d85517b18402810c4","_dom_classes":[],"description":"","_model_name":"FloatProgressModel","bar_style":"","max":100,"_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","value":0,"_view_count":null,"_view_module_version":"1.5.0","orientation":"horizontal","min":0,"description_tooltip":null,"_model_module":"@jupyter-widgets/controls","layout":"IPY_MODEL_0e066cc31db9454fa555761fadc0d706"}},"18098799052e443baaa6c9e27eaac950":{"model_module":"@jupyter-widgets/controls","model_name":"HTMLModel","model_module_version":"1.5.0","state":{"_view_name":"HTMLView","style":"IPY_MODEL_6638c74679304aa98eb041d7e8a4b6f3","_dom_classes":[],"description":"","_model_name":"HTMLModel","placeholder":"​","_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","value":" 0/100 [00:00&lt;?, ?epoch/s]","_view_count":null,"_view_module_version":"1.5.0","description_tooltip":null,"_model_module":"@jupyter-widgets/controls","layout":"IPY_MODEL_1d5e85caac5d40789bad8fd3cd584dbc"}},"48ac18cc60b7457db81eee6a2e4984a5":{"model_module":"@jupyter-widgets/controls","model_name":"DescriptionStyleModel","model_module_version":"1.5.0","state":{"_view_name":"StyleView","_model_name":"DescriptionStyleModel","description_width":"","_view_module":"@jupyter-widgets/base","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.2.0","_model_module":"@jupyter-widgets/controls"}},"46e55f0c552547d4b543b1ccfb0ac2b7":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","model_module_version":"1.2.0","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}},"6f25657095c4435d85517b18402810c4":{"model_module":"@jupyter-widgets/controls","model_name":"ProgressStyleModel","model_module_version":"1.5.0","state":{"_view_name":"StyleView","_model_name":"ProgressStyleModel","description_width":"","_view_module":"@jupyter-widgets/base","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.2.0","bar_color":null,"_model_module":"@jupyter-widgets/controls"}},"0e066cc31db9454fa555761fadc0d706":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","model_module_version":"1.2.0","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}},"6638c74679304aa98eb041d7e8a4b6f3":{"model_module":"@jupyter-widgets/controls","model_name":"DescriptionStyleModel","model_module_version":"1.5.0","state":{"_view_name":"StyleView","_model_name":"DescriptionStyleModel","description_width":"","_view_module":"@jupyter-widgets/base","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.2.0","_model_module":"@jupyter-widgets/controls"}},"1d5e85caac5d40789bad8fd3cd584dbc":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","model_module_version":"1.2.0","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}}}}},"cells":[{"cell_type":"code","source":["!pip install -q grape -U\n","!pip install -q plot_keras_history seedir silence_tensorflow\n","# !pip install -q tsnecuda==3.0.0+cu110 -f https://tsnecuda.isx.ai/tsnecuda_stable.html --no-dependencies\n","!pip install -q faiss"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"wIRPodcb93UO","executionInfo":{"status":"ok","timestamp":1646432881541,"user_tz":480,"elapsed":57602,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"fa331263-8e91-476e-bbb8-cc482095ef76"},"execution_count":1,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[K     |████████████████████████████████| 97 kB 3.3 MB/s \n","\u001b[K     |████████████████████████████████| 99 kB 8.8 MB/s \n","\u001b[?25h  Building wheel for grape (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for embiggen (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for cache-decorator (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for compress-json (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for ddd-subplots (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for dict-hash (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for deflate-dict (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for extra-keras-metrics (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for keras-mixed-sequence (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for sanitize-ml-labels (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for downloaders (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for environments-utils (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for py-cpuinfo (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for pygifsicle (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for userinput (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for IPy (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for validate-email (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for validate-version-code (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[K     |████████████████████████████████| 112 kB 5.4 MB/s \n","\u001b[K     |████████████████████████████████| 174 kB 42.9 MB/s \n","\u001b[?25h  Building wheel for plot-keras-history (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for silence-tensorflow (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for emoji (setup.py) ... \u001b[?25l\u001b[?25hdone\n"]}]},{"cell_type":"code","source":["!wget https://kg-hub.berkeleybop.io/kg-ontoml/20220304/KG-OntoML.tar.gz"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"NwOGSINU8zwR","executionInfo":{"status":"ok","timestamp":1646432882288,"user_tz":480,"elapsed":757,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"06151636-23a0-4a3b-e387-2f0fe122f010"},"execution_count":2,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-03-04 22:28:01--  https://kg-hub.berkeleybop.io/kg-ontoml/20220304/KG-OntoML.tar.gz\n","Resolving kg-hub.berkeleybop.io (kg-hub.berkeleybop.io)... 108.156.91.115, 108.156.91.16, 108.156.91.114, ...\n","Connecting to kg-hub.berkeleybop.io (kg-hub.berkeleybop.io)|108.156.91.115|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 44235554 (42M) [application/gzip]\n","Saving to: ‘KG-OntoML.tar.gz.1’\n","\n","KG-OntoML.tar.gz.1  100%[===================>]  42.19M  78.9MB/s    in 0.5s    \n","\n","2022-03-04 22:28:02 (78.9 MB/s) - ‘KG-OntoML.tar.gz.1’ saved [44235554/44235554]\n","\n"]}]},{"cell_type":"code","source":["!tar -xvzf KG-OntoML.tar.gz"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"6UPyUTi__jiT","executionInfo":{"status":"ok","timestamp":1646432885316,"user_tz":480,"elapsed":3033,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"c68067bb-2b10-4b8d-d14c-8d9a0ba837d6"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["merged-kg_nodes.tsv\n","merged-kg_edges.tsv\n"]}]},{"cell_type":"code","source":["!head merged-kg_nodes.tsv"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5b5RAh9gBdWM","executionInfo":{"status":"ok","timestamp":1646432885470,"user_tz":480,"elapsed":160,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"a291c9e1-65d3-4532-af89-c6e2816a66e2"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["id\tcategory\tname\tdescription\txref\tprovided_by\tsynonym\t0000052\t0000061\t0000114\t0000178\t0000184\t0000231\t0000233\t0000234\t0000386\t0000399\t0000424\t0000425\t0000426\t0000589\t0000905\t0006012\t0037192\t0100001\t9000002\t:http://attempto.ifi.uzh.ch/ace_lexicon#TV_pl\t:http://attempto.ifi.uzh.ch/ace_lexicon#TV_sg\t:http://attempto.ifi.uzh.ch/ace_lexicon#TV_vbg\t:http://geneontology.org/formats/oboInOwl#created_by\t:http://purl.obolibrary.org/obo/chebi/charge\t:http://purl.obolibrary.org/obo/chebi/formula\t:http://purl.obolibrary.org/obo/chebi/inchi\t:http://purl.obolibrary.org/obo/chebi/inchikey\t:http://purl.obolibrary.org/obo/chebi/mass\t:http://purl.obolibrary.org/obo/chebi/monoisotopicmass\t:http://purl.obolibrary.org/obo/chebi/smiles\t:http://purl.org/spar/cito/citesAsAuthority\t:http://usefulinc.com/ns/doap#revision\t:http://www.ebi.ac.uk/cellline/definition\t:http://www.ebi.ac.uk/cellline/definition_editor\t:http://www.ebi.ac.uk/efo/alternative_term\t:http://www.ebi.ac.uk/efo/definition\t:http://www.ebi.ac.uk/efo/definition_citation\t:http://www.ebi.ac.uk/efo/definition_editor\t:http://www.ebi.ac.uk/efo/example_of_usage\t:http://www.w3.org/2003/06/sw-vocab-status/ns#term_status\t:https://w3id.org/biodatamodels/gff/end\t:https://w3id.org/biodatamodels/gff/start\t:ttp://www.geneontology.org/formats/oboInOwl#created_by\tBFO_CLIF_specification_label\tBFO_OWL_specification_label\tDate\tUri\talt_id\talternative_term\tauthor\tcl#created_by\tclose_match\tcomment\tconsider\tcontributor\tcore#provenance_notes\tcreated_by\tcreation_date\tcreator\tcurator_note\tdefinition_source\tdepicted_by\tdeprecated\teditor_note\teditor_preferred_term\telucidation\texample_of_usage\tgenome_build\thas_alternative_id\thas_associated_axiom(fol)\thas_associated_axiom(nl)\thas_o_b_o_namespace\thas_synonym\thsapdv#end_dpf\thsapdv#end_mpb\thsapdv#end_ypb\thsapdv#start_dpf\thsapdv#start_mpb\thsapdv#start_ypb\timported_from\tin_subset\tis_a_defining_property_chain_axiom\tis_class_level\tis_cyclic\tis_defined_by\tis_metadata_tag\tis_transitive\tissued\tknowledge_source\tlogical_interpretation\tlogical_macro_assertion_on_an_annotation_property\tmodified\tmondo#excluded_synonym\tmondo#pathogenesis\tmondo#related\tnote\tobject\tpage\tpredicate\tpresent_in_taxon\tprior_version\trelation\tsee_also\tshorthand\tsource\tsubject\tsubset\tterm_editor\ttype\tu_b_p_r_o_p_0000001\tu_b_p_r_o_p_0000002\tu_b_p_r_o_p_0000003\tu_b_p_r_o_p_0000004\tu_b_p_r_o_p_0000005\tu_b_p_r_o_p_0000006\tu_b_p_r_o_p_0000007\tu_b_p_r_o_p_0000008\tu_b_p_r_o_p_0000009\tu_b_p_r_o_p_0000010\tu_b_p_r_o_p_0000011\tu_b_p_r_o_p_0000012\tu_b_p_r_o_p_0000013\tu_b_p_r_o_p_0000014\tu_b_p_r_o_p_0000015\tu_b_p_r_o_p_0000103\tu_b_p_r_o_p_0000104\tu_b_p_r_o_p_0000105\tu_b_p_r_o_p_0000106\tu_b_p_r_o_p_0000107\tu_b_p_r_o_p_0000108\tu_b_p_r_o_p_0000111\tu_b_p_r_o_p_0000112\n","OBO:FBbt_00004566\tbiolink:NamedThing\tDrosophila prothoracic episternum\t\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","OBO:FBbt_00004554\tbiolink:NamedThing\tDrosophila episternum\t\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","XPO:0136009\tbiolink:NamedThing\tincreased cell population proliferation in forelimb zeugopod\tAn increased occurrence of cell population proliferation in forelimb zeugopod.\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","XPO:0135498\tbiolink:NamedThing\tincreased cell population proliferation in forelimb\tAn increased occurrence of cell population proliferation in forelimb.\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","CHEBI:140678\tbiolink:NamedThing\t4,4-dihydroxy-all-trans-beta-carotene\t\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","CHEBI:35163\tbiolink:NamedThing\tcyclic carotene\t\t\tGraph\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","GO:0045959\tbiolink:NamedThing\tnegative regulation of complement activation, classical pathway\tAny process that stops, prevents, or reduces the frequency, rate or extent of complement activation by the classical pathway.\t\tGraph\tnegative regulation of complement cascade, classical pathway|downregulation of complement activation, classical pathway|inhibition of complement activation, classical pathway|down regulation of complement activation, classical pathway|down-regulation of complement activation, classical pathway\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\tbiological_process\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","GO:0045916\tbiolink:NamedThing\tnegative regulation of complement activation\tAny process that stops, prevents, or reduces the frequency, rate or extent of complement activation.\t\tGraph\tinhibition of complement activation|downregulation of complement activation|down regulation of complement activation|negative regulation of complement cascade|down-regulation of complement activation\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\tbiological_process\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n","OBO:ZFA_0005195\tbiolink:NamedThing\tMiD3cl\tHindbrain interneuron, part of rhombomere 6, with one caudal-lateral neuron in each hemi-rhombomere.  The fibers pass through the lateral longitudinal fasciculus.\t\tGraph\tmiddle dorsal 3 contralateral llf interneuron \t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\tTAO:0005195\tThis term is a leaf node in the zebrafish AO and is not in the CL.\t\t\t\tCeri\t2009-06-22T05:10:40Z\t\t\t\t\t\t\t\t\t\t\t\t\t\tzebrafish_anatomy\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\towl:Class\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\n"]}]},{"cell_type":"code","source":["!head merged-kg_edges.tsv"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"hSQXmOLvBqfA","executionInfo":{"status":"ok","timestamp":1646432885609,"user_tz":480,"elapsed":143,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"53d1086b-7177-4823-a258-5f7a91c7348e"},"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["id\tsubject\tpredicate\tobject\tcategory\trelation\tknowledge_source\tlogical_interpretation\n","OBO:FBbt_00004566-biolink:subclass_of-OBO:FBbt_00004554\tOBO:FBbt_00004566\tbiolink:subclass_of\tOBO:FBbt_00004554\t\trdfs:subClassOf\tGraph\t\n","urn:uuid:a564b8a3-6fee-45f8-8cc8-d0d6093c6c64\tOBO:FBbt_00004566\tbiolink:develops_from\tOBO:FBbt_00001781\tbiolink:Association\tRO:0002202\tGraph\towlstar:AllSomeInterpretation\n","urn:uuid:c1ef3711-892c-4470-b9b0-062fc0c95fda\tOBO:FBbt_00004566\tbiolink:part_of\tOBO:FBbt_00004565\tbiolink:Association\tBFO:0000050\tGraph\towlstar:AllSomeInterpretation\n","OBO:FBbt_00004554-biolink:subclass_of-OBO:FBbt_00004475\tOBO:FBbt_00004554\tbiolink:subclass_of\tOBO:FBbt_00004475\t\trdfs:subClassOf\tGraph\t\n","urn:uuid:1f6a4aa4-abee-4335-acef-cab6be117d77\tOBO:FBbt_00004554\tbiolink:part_of\tOBO:FBbt_00004553\tbiolink:Association\tBFO:0000050\tGraph\towlstar:AllSomeInterpretation\n","XPO:0136009-biolink:subclass_of-XPO:0135498\tXPO:0136009\tbiolink:subclass_of\tXPO:0135498\t\trdfs:subClassOf\tGraph\t\n","XPO:0136009-biolink:subclass_of-XPO:0135978\tXPO:0136009\tbiolink:subclass_of\tXPO:0135978\t\trdfs:subClassOf\tGraph\t\n","urn:uuid:c9f9bc04-b539-4d30-98b7-3b06366f037e\tXPO:0136009\tbiolink:has_part\tNdcb697a53b974cb08f87d5ece7463466\tbiolink:Association\tBFO:0000051\tGraph\towlstar:AllSomeInterpretation\n","XPO:0135498-biolink:subclass_of-XPO:0135496\tXPO:0135498\tbiolink:subclass_of\tXPO:0135496\t\trdfs:subClassOf\tGraph\t\n"]}]},{"cell_type":"code","source":["# In order to disable often useless TensorFlow warnings\n","import silence_tensorflow.auto"],"metadata":{"id":"4DQwAU_l_nHr","executionInfo":{"status":"ok","timestamp":1646432891015,"user_tz":480,"elapsed":5410,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}}},"execution_count":6,"outputs":[]},{"cell_type":"code","source":["import ensmallen\n","from ensmallen import Graph\n","import embiggen"],"metadata":{"id":"jFpRMzdI_yD9","executionInfo":{"status":"ok","timestamp":1646432896679,"user_tz":480,"elapsed":5679,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}}},"execution_count":7,"outputs":[]},{"cell_type":"code","source":["g= Graph.from_csv(\n","  directed=False,\n","  node_path='merged-kg_nodes.tsv',\n","  edge_path='merged-kg_edges.tsv',\n","  verbose=True,\n","  nodes_column='id',\n","  node_list_node_types_column='category',\n","  default_node_type='biolink:NamedThing',\n","  sources_column='subject',\n","  destinations_column='object',\n","  default_edge_type='biolink:related_to'\n",")"],"metadata":{"id":"t7EM5bJr_8tJ","executionInfo":{"status":"ok","timestamp":1646432898496,"user_tz":480,"elapsed":113,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}}},"execution_count":8,"outputs":[]},{"cell_type":"code","source":["g"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":711},"id":"bUslR8PPA5Jb","executionInfo":{"status":"ok","timestamp":1646432899118,"user_tz":480,"elapsed":120,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"dbe2626f-318f-447c-bb7a-308a95d82295"},"execution_count":9,"outputs":[{"output_type":"execute_result","data":{"text/html":["<p>The undirected graph has 511960 homogeneous nodes and 766323 homogeneous edges.</p><h3>RAM requirements</h3><p>The RAM requirements for the nodes and edges data structures are 129.83MB and 3.73MB respectively.</p><h3>Degree centrality</h3><p>The minimum node degree is 0, the maximum node degree is 21699, the mode degree is 1, the mean degree is 2.99 and the node degree median is 1.</p><p>The nodes with highest degree centrality are: <a href='http://purl.obolibrary.org/obo/CLO_0000374' target='_blank' title='Go to Cell Line Ontology to get more informations about CLO:0000374'>CLO:0000374</a> (degree 21699 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), GOP:chebi_ph7_3 (degree 11295 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='https://www.ncbi.nlm.nih.gov/taxonomy/?term=species' target='_blank' title='Go to NCBI Taxonomy to get more informations about NCBITaxon:species'>NCBITaxon:species</a> (degree 5014 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/IAO_0000227' target='_blank' title='Go to Information Artifact Ontology to get more informations about IAO:0000227'>IAO:0000227</a> (degree 4510 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://purl.obolibrary.org/obo/mondo#ordo_disease' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:mondo#ordo_disease'>OBO:mondo#ordo_disease</a> (degree 3981 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>).</p><h3>Disconnected nodes</h3><p>Disconnected nodes are nodes that are not connected to any other node.The graph contains 46924 disconnected nodes.</p><h4>Singleton nodes</h4><p>Singleton nodes are nodes with no edge to other nodes nor selfloops. The graph contains 46922 singleton nodes, which are <a href='http://purl.obolibrary.org/obo/GENO_0000634' target='_blank' title='Go to Genotype Ontology to get more informations about GENO:0000634'>GENO:0000634</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/GOREL_0002005' target='_blank' title='Go to Gene Ontology Relations to get more informations about GOREL:0002005'>GOREL:0002005</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/CLO_0000167' target='_blank' title='Go to Cell Line Ontology to get more informations about CLO:0000167'>CLO:0000167</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://www.obofoundry.org/ro/ro.owl#contained_in' target='_blank' title='Go to website to get more informations about http:&#x2f;&#x2f;www.obofoundry.org&#x2f;ro&#x2f;ro.owl#contained_in'>http:&#x2f;&#x2f;www.obofoundry.org&#x2f;ro&#x2f;ro.owl#contained_in</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://purl.obolibrary.org/obo/mp/mp-equivalent-axioms-subq#has_quality' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:mp&#x2f;mp-equivalent-axioms-subq#has_quality'>OBO:mp&#x2f;mp-equivalent-axioms-subq#has_quality</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), plus other 46917 singleton nodes.</p><h4>Singleton nodes with selfloops</h4><p>Singleton nodes with selfloops are nodes with no edge to other nodes and have exclusively selfloops. The graph contains 2 singleton nodes with selfloops, which are <a href='http://purl.obolibrary.org/obo/go/extensions/ro_pending#results_in_connection_of' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:go&#x2f;extensions&#x2f;ro_pending#results_in_connection_of'>OBO:go&#x2f;extensions&#x2f;ro_pending#results_in_connection_of</a> (degree 1 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://xmlns.com/foaf/0.1/depicted_by' target='_blank' title='Go to Friend of a Friend to get more informations about foaf:depicted_by'>foaf:depicted_by</a> (degree 1 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>).</p><h3>Node types</h3><p>The graph has a single node type, which is <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>. Note that this means that all nodes have the same node type, that is, all nodes are homogeneous..</p><h4>RAM requirements</h4><p>The RAM requirements for the node types data structure is 25.60MB.</p><h3>Edge types</h3><p>The graph has a single edge type, which is <a href='https://biolink.github.io/biolink-model/docs/related_to.html' target='_blank' title='Go to BioLink to get more informations about biolink:related_to'>biolink:related_to</a>. Note that this means that all edges have the same edge type, that is, all edges are homogeneous..</p><h4>RAM requirements</h4><p>The RAM requirements for the edge types data structure is 6.13MB.</p>"],"text/plain":["<p>The undirected graph has 511960 homogeneous nodes and 766323 homogeneous edges.</p><h3>RAM requirements</h3><p>The RAM requirements for the nodes and edges data structures are 129.83MB and 3.73MB respectively.</p><h3>Degree centrality</h3><p>The minimum node degree is 0, the maximum node degree is 21699, the mode degree is 1, the mean degree is 2.99 and the node degree median is 1.</p><p>The nodes with highest degree centrality are: <a href='http://purl.obolibrary.org/obo/CLO_0000374' target='_blank' title='Go to Cell Line Ontology to get more informations about CLO:0000374'>CLO:0000374</a> (degree 21699 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), GOP:chebi_ph7_3 (degree 11295 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='https://www.ncbi.nlm.nih.gov/taxonomy/?term=species' target='_blank' title='Go to NCBI Taxonomy to get more informations about NCBITaxon:species'>NCBITaxon:species</a> (degree 5014 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/IAO_0000227' target='_blank' title='Go to Information Artifact Ontology to get more informations about IAO:0000227'>IAO:0000227</a> (degree 4510 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://purl.obolibrary.org/obo/mondo#ordo_disease' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:mondo#ordo_disease'>OBO:mondo#ordo_disease</a> (degree 3981 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>).</p><h3>Disconnected nodes</h3><p>Disconnected nodes are nodes that are not connected to any other node.The graph contains 46924 disconnected nodes.</p><h4>Singleton nodes</h4><p>Singleton nodes are nodes with no edge to other nodes nor selfloops. The graph contains 46922 singleton nodes, which are <a href='http://purl.obolibrary.org/obo/GENO_0000634' target='_blank' title='Go to Genotype Ontology to get more informations about GENO:0000634'>GENO:0000634</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/GOREL_0002005' target='_blank' title='Go to Gene Ontology Relations to get more informations about GOREL:0002005'>GOREL:0002005</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://purl.obolibrary.org/obo/CLO_0000167' target='_blank' title='Go to Cell Line Ontology to get more informations about CLO:0000167'>CLO:0000167</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), <a href='http://www.obofoundry.org/ro/ro.owl#contained_in' target='_blank' title='Go to website to get more informations about http:&#x2f;&#x2f;www.obofoundry.org&#x2f;ro&#x2f;ro.owl#contained_in'>http:&#x2f;&#x2f;www.obofoundry.org&#x2f;ro&#x2f;ro.owl#contained_in</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://purl.obolibrary.org/obo/mp/mp-equivalent-axioms-subq#has_quality' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:mp&#x2f;mp-equivalent-axioms-subq#has_quality'>OBO:mp&#x2f;mp-equivalent-axioms-subq#has_quality</a> (node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>), plus other 46917 singleton nodes.</p><h4>Singleton nodes with selfloops</h4><p>Singleton nodes with selfloops are nodes with no edge to other nodes and have exclusively selfloops. The graph contains 2 singleton nodes with selfloops, which are <a href='http://purl.obolibrary.org/obo/go/extensions/ro_pending#results_in_connection_of' target='_blank' title='Go to Open Biomedical Ontology to get more informations about OBO:go&#x2f;extensions&#x2f;ro_pending#results_in_connection_of'>OBO:go&#x2f;extensions&#x2f;ro_pending#results_in_connection_of</a> (degree 1 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>) and <a href='http://xmlns.com/foaf/0.1/depicted_by' target='_blank' title='Go to Friend of a Friend to get more informations about foaf:depicted_by'>foaf:depicted_by</a> (degree 1 and node type <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>).</p><h3>Node types</h3><p>The graph has a single node type, which is <a href='https://biolink.github.io/biolink-model/docs/NamedThing.html' target='_blank' title='Go to BioLink to get more informations about biolink:NamedThing'>biolink:NamedThing</a>. Note that this means that all nodes have the same node type, that is, all nodes are homogeneous..</p><h4>RAM requirements</h4><p>The RAM requirements for the node types data structure is 25.60MB.</p><h3>Edge types</h3><p>The graph has a single edge type, which is <a href='https://biolink.github.io/biolink-model/docs/related_to.html' target='_blank' title='Go to BioLink to get more informations about biolink:related_to'>biolink:related_to</a>. Note that this means that all edges have the same edge type, that is, all edges are homogeneous..</p><h4>RAM requirements</h4><p>The RAM requirements for the edge types data structure is 6.13MB.</p>"]},"metadata":{},"execution_count":9}]},{"cell_type":"code","source":["from embiggen.pipelines import compute_node_embedding\n","from plot_keras_history import plot_history\n","\n","node_embedding_method_name = \"SkipGram\"\n","\n","first_order_rw_node_embedding, training_history = compute_node_embedding(\n","    g,\n","    node_embedding_method_name=node_embedding_method_name,\n",")\n","\n","plot_history(\n","    training_history,\n","    title=\"First-order random walk based {} model applied to graph {}\".format(\n","        node_embedding_method_name,\n","        g.get_name()\n","    )\n",")"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":439,"referenced_widgets":["fc7b9510580d440e8e226aee783a9ee3","6def71a8cb2d4717a6abf73d4f8e5265","ca9a3f2486ae423b900c6f55355f654a","30b7b7b3d0aa4fe49ba756cfce0254e2","18098799052e443baaa6c9e27eaac950","48ac18cc60b7457db81eee6a2e4984a5","46e55f0c552547d4b543b1ccfb0ac2b7","6f25657095c4435d85517b18402810c4","0e066cc31db9454fa555761fadc0d706","6638c74679304aa98eb041d7e8a4b6f3","1d5e85caac5d40789bad8fd3cd584dbc"]},"id":"eO4nH5nKB_jK","executionInfo":{"status":"error","timestamp":1646433401949,"user_tz":480,"elapsed":502843,"user":{"displayName":"Justin Reese","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GiOosE2s-Zd2j7gWtW3wl603baopxDZOj-_wUfYLQ=s64","userId":"05932852198201752265"}},"outputId":"4ab9c7a8-49e5-4fe1-8f67-f727be6319b9"},"execution_count":10,"outputs":[{"output_type":"display_data","data":{"application/vnd.jupyter.widget-view+json":{"model_id":"fc7b9510580d440e8e226aee783a9ee3","version_minor":0,"version_major":2},"text/plain":["0epoch [00:00, ?epoch/s]"]},"metadata":{}},{"output_type":"error","ename":"KeyboardInterrupt","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)","\u001b[0;32m<ipython-input-10-08ee002667fe>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m first_order_rw_node_embedding, training_history = compute_node_embedding(\n\u001b[1;32m      7\u001b[0m     \u001b[0mg\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mnode_embedding_method_name\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnode_embedding_method_name\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m )\n\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/embiggen/pipelines/compute_node_embedding.py\u001b[0m in \u001b[0;36mcompute_node_embedding\u001b[0;34m(graph, node_embedding_method_name, use_mirrored_strategy, devices, fit_kwargs, verbose, automatically_drop_unsupported_parameters, automatically_enable_time_memory_tradeoffs, automatically_sort_by_decreasing_outbound_node_degree, **kwargs)\u001b[0m\n\u001b[1;32m    345\u001b[0m         \u001b[0muse_mirrored_strategy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0muse_mirrored_strategy\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    346\u001b[0m         \u001b[0mdevices\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdevices\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 347\u001b[0;31m         \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    348\u001b[0m     )\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/cache_decorator/cache.py\u001b[0m in \u001b[0;36mwrapped\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    475\u001b[0m             \u001b[0;31m# otherwise compute the result\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    476\u001b[0m             \u001b[0mstart_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 477\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunction\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    478\u001b[0m             \u001b[0mend_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    479\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/embiggen/pipelines/compute_node_embedding.py\u001b[0m in \u001b[0;36m_compute_node_embedding\u001b[0;34m(graph, graph_name, node_embedding_method_name, fit_kwargs, verbose, use_mirrored_strategy, devices, **kwargs)\u001b[0m\n\u001b[1;32m    178\u001b[0m         \u001b[0mstrategy\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdistribute\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMirroredStrategy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdevices\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdevices\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    179\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mstrategy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscope\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 180\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0m_train_model\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    181\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0m_train_model\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    182\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/embiggen/pipelines/compute_node_embedding.py\u001b[0m in \u001b[0;36m_train_model\u001b[0;34m(graph, node_embedding_method_name, fit_kwargs, verbose, support_mirrored_strategy, **kwargs)\u001b[0m\n\u001b[1;32m    111\u001b[0m     history = model.fit(\n\u001b[1;32m    112\u001b[0m         \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 113\u001b[0;31m         \u001b[0;34m**\u001b[0m\u001b[0mfit_kwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    114\u001b[0m     )\n\u001b[1;32m    115\u001b[0m     \u001b[0;31m# Extracting computed embedding\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/embiggen/embedders/node2vec.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, epochs, early_stopping_monitor, early_stopping_min_delta, early_stopping_patience, early_stopping_mode, reduce_lr_monitor, reduce_lr_min_delta, reduce_lr_patience, reduce_lr_mode, reduce_lr_factor, verbose, **kwargs)\u001b[0m\n\u001b[1;32m    223\u001b[0m             \u001b[0mreduce_lr_factor\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mreduce_lr_factor\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    224\u001b[0m             \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 225\u001b[0;31m             \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    226\u001b[0m         )\n\u001b[1;32m    227\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/embiggen/embedders/embedder.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, early_stopping_min_delta, early_stopping_patience, reduce_lr_min_delta, reduce_lr_patience, epochs, early_stopping_monitor, early_stopping_mode, reduce_lr_monitor, reduce_lr_mode, reduce_lr_factor, verbose, *args, **kwargs)\u001b[0m\n\u001b[1;32m    359\u001b[0m                 \u001b[0;34m*\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    360\u001b[0m             ],\n\u001b[0;32m--> 361\u001b[0;31m             \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    362\u001b[0m         ).history)\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/utils/traceback_utils.py\u001b[0m in \u001b[0;36merror_handler\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m       \u001b[0;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     65\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# pylint: disable=broad-except\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m       \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_process_traceback_frames\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__traceback__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/engine/training.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)\u001b[0m\n\u001b[1;32m   1382\u001b[0m                 _r=1):\n\u001b[1;32m   1383\u001b[0m               \u001b[0mcallbacks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mon_train_batch_begin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstep\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1384\u001b[0;31m               \u001b[0mtmp_logs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtrain_function\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0miterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1385\u001b[0m               \u001b[0;32mif\u001b[0m \u001b[0mdata_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshould_sync\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1386\u001b[0m                 \u001b[0mcontext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masync_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/util/traceback_utils.py\u001b[0m in \u001b[0;36merror_handler\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    148\u001b[0m     \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    149\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m       \u001b[0;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    151\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m       \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_process_traceback_frames\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__traceback__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/def_function.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, *args, **kwds)\u001b[0m\n\u001b[1;32m    913\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    914\u001b[0m       \u001b[0;32mwith\u001b[0m \u001b[0mOptionalXlaContext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_jit_compile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 915\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    916\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    917\u001b[0m       \u001b[0mnew_tracing_count\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexperimental_get_tracing_count\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/def_function.py\u001b[0m in \u001b[0;36m_call\u001b[0;34m(self, *args, **kwds)\u001b[0m\n\u001b[1;32m    945\u001b[0m       \u001b[0;31m# In this case we have created variables on the first call, so we run the\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    946\u001b[0m       \u001b[0;31m# defunned version which is guaranteed to never create variables.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 947\u001b[0;31m       \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_stateless_fn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# pylint: disable=not-callable\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    948\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_stateful_fn\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    949\u001b[0m       \u001b[0;31m# Release the lock early so that multiple threads can perform the call\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   2955\u001b[0m        filtered_flat_args) = self._maybe_define_function(args, kwargs)\n\u001b[1;32m   2956\u001b[0m     return graph_function._call_flat(\n\u001b[0;32m-> 2957\u001b[0;31m         filtered_flat_args, captured_inputs=graph_function.captured_inputs)  # pylint: disable=protected-access\n\u001b[0m\u001b[1;32m   2958\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2959\u001b[0m   \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\u001b[0m in \u001b[0;36m_call_flat\u001b[0;34m(self, args, captured_inputs, cancellation_manager)\u001b[0m\n\u001b[1;32m   1852\u001b[0m       \u001b[0;31m# No tape is watching; skip to running the function.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1853\u001b[0m       return self._build_call_outputs(self._inference_function.call(\n\u001b[0;32m-> 1854\u001b[0;31m           ctx, args, cancellation_manager=cancellation_manager))\n\u001b[0m\u001b[1;32m   1855\u001b[0m     forward_backward = self._select_forward_and_backward_functions(\n\u001b[1;32m   1856\u001b[0m         \u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\u001b[0m in \u001b[0;36mcall\u001b[0;34m(self, ctx, args, cancellation_manager)\u001b[0m\n\u001b[1;32m    502\u001b[0m               \u001b[0minputs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    503\u001b[0m               \u001b[0mattrs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mattrs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 504\u001b[0;31m               ctx=ctx)\n\u001b[0m\u001b[1;32m    505\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m           outputs = execute.execute_with_cancellation(\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/execute.py\u001b[0m in \u001b[0;36mquick_execute\u001b[0;34m(op_name, num_outputs, inputs, attrs, ctx, name)\u001b[0m\n\u001b[1;32m     53\u001b[0m     \u001b[0mctx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mensure_initialized\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m     tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\n\u001b[0;32m---> 55\u001b[0;31m                                         inputs, attrs, num_outputs)\n\u001b[0m\u001b[1;32m     56\u001b[0m   \u001b[0;32mexcept\u001b[0m \u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_NotOkStatusException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}]}]}